src/components/cdPipeline/CDPipeline.tsx

```diff
@@ -283,15 +292,65 @@ export default function CDPipeline({
         setAvailableTags(tags)
     }
 
-    useEffect(() => {
-        getInit()
-    }, [])
+    const handlePopulatePluginDataStore = async (form: PipelineFormType) => {
+        if (!form) {
+            return
+        }
 
-    useEffect(() => {
-        if (formData.environmentId) {
-            getConfigMapSecrets()
+        const preBuildPluginIds = getPluginIdsFromBuildStage(form.preBuildStage)
+        const postBuildPluginIds = getPluginIdsFromBuildStage(form.postBuildStage)
+        const uniquePluginIds = Array.from(new Set([...preBuildPluginIds, ...postBuildPluginIds]))
+
+        if (processPluginData) {
+            await getMandatoryPluginData(form, uniquePluginIds)
+            return
+        }
+
+        if (uniquePluginIds.length === 0) {
+            return
         }
```

Comment:
shouldn't the length check be first?

Reply:
There can be no steps yet requirement of mandatory plugins